server/app/views/questiontypes/DateQuestionFragment.html

```diff
@@ -0,0 +1,155 @@
+<div
+  th:fragment="date (question, questionRendererParams)"
+  class="cf-question-date"
+>
+  <fieldset
+    class="usa-fieldset"
+    th:with="questionId=${#strings.randomAlphanumeric(8)},
+             dateQuestion=${question.createDateQuestion()},
+             firstPathWithError=${dateQuestion.getFirstPathWithError()}"
+  >
+    <!-- Title and Help Text -->
+    <div
+      th:replace="~{questiontypes/QuestionBaseFragment :: titleAndHelpTextMultipleInput(${question}, ${questionId})}"
+    ></div>
+
+    <!-- Display errors -->
+    <div
+      th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${dateQuestion.getValidationErrors().get(question.getContextualizedPath())})}"
+    ></div>
+    <!-- Display error messages for the shared date path. We actually use 3 different paths for
+    inputs, but all errors are reported on the date path. -->
+    <div
+      th:replace="~{questiontypes/QuestionBaseFragment :: validationErrors(${dateQuestion.getValidationErrors().get(dateQuestion.getDatePath())})}"
+    ></div>
+
+    <!-- Hidden input to delete the old single date value. -->
+    <input hidden th:name="${dateQuestion.getDatePath()}" value="" />
+    <div class="usa-memorable-date">
+      <!-- Month Input -->
+      <div
+        class="usa-form-group usa-form-group--month usa-form-group--select cf-date-month"
+        th:with="monthId=${#strings.randomAlphanumeric(8)},monthPath=${dateQuestion.getMonthPath()}"
```

Comment:
(question, blocking) Does the `strings.randomAlphanumeric` get used as the element's ID? If yes, there should be some alpha only prefix to [avoid this problem](https://github.com/civiform/civiform/pull/6297).

Reply:
it does get used as the id of the html element. From what I can see we only currently do this for 1) the form element on the program predicates edit view, and 2) modals.

Would you mind if I go though in a future PR and update all of the instances where it's used in the thymeleaf implementation, since it's used kind of everywhere and I don't want to bloat up this PR?

Reply:
Yeah, that's fine. Best to do them all at once so none get missed.